test/scripts/e2e_client_runner.py

```diff
@@ -264,12 +264,39 @@ def wait(self, timeout):
                 self.ok = False
                 self._terminate()
 
-def goal_network_stop(netdir):
+
+# 'network stop' and 'network delete' are also tested and used as cleanup procedures
+# so it re-raises exception in 'test' mode
+already_stopped = False
+already_deleted = False
+
+def goal_network_stop(netdir, normal_cleanup=False):
+    global already_stopped, already_deleted
+    if already_stopped or already_deleted:
+        return
+
     logger.info('stop network in %s', netdir)
     try:
         xrun(['goal', 'network', 'stop', '-r', netdir], timeout=10)
     except Exception as e:
         logger.error('error stopping network', exc_info=True)
+        if normal_cleanup:
+            raise e
+    already_stopped = True
+
+def goal_network_delete(netdir, normal_cleanup=False):
```

Comment:
add atexit registration of this after `goal network create` ?

Reply:
This is probably not needed because there is a generic cleaner with `rmtree`.
`already_deleted` flag is still needed to prevent running `goal_network_stop` from atexit handler.